spec/factories/chunks.rb

```diff
@@ -1,13 +1,13 @@
 FactoryBot.define do
   factory :chunk do
-    upload
+    upload { create(:upload, :skip_validation) }
```

Comment:
Whenever tests use allow(object).to receive(:storage_provider).and_return(mocked_storage_provider) (or expect..) on an Upload or Chunk, the validations will still fail because the object is being created before the mock is applied. This is especially challenging for objects that create (objects that create) uploads.